### PR TITLE
Add a footer component

### PR DESCRIPTION
**Pull request by lal65**
# Description:

Add a footer component and certain smaller helpers to reduce boilerplate.

## Footer

Provides the overall layout and content slots:

*   top first
*   top second
*   top third
*   middle first
*   middle second
*   middle third
*   colophon

Provides one customization point: _**bottom\_background**_ which can be overridden to 'white'.

## List

Provides a generic list component with certain variables:

ordered - boolean - controls list semantic

orientation - string - can be 'horizontal' for a horizontal list

separator - string - can be 'dot' to separate items with dots.

density - string - can be 'loose' to separate items with more space

## Physical Address

Provides an element for rendering a physical address with certain variables:

semantic - boolean - if true, renders within an _**\<address>**_ element

name - string - optional name on the address

street - string - optional street / building code

locality - string - locality of address

state - string - state of address

postcode - string - postal code of address

## New sprite feature: expand

We seem to target sprites like _**.component > .sprite**_ quite a bit in order to set explicit dimensions.  A better approach is for the sprite to delegate said controls to its context.  Introducing the _**expand**_ modifier for sprites.

### Example

```twig
<div class="my-component">
  <div class="my-component__sprite">
    {% include '@psu-ooe/sprite/sprite.twig' with { name: 'psu-mark' } only %}
  </div>
</div>
```

Typically, to give the sprite explicit sizing, one would do this...

```css
.my-component {
  &__sprite {
    .sprite {
      width: 20rem;
      height: auto;
    }
  }
}
```

...but with _**expand**_, one can now do this:

```twig
<div class="my-component">
  <div class="my-component__sprite">
    {% include '@psu-ooe/sprite/sprite.twig' with { name: 'psu-mark', expand: true } only %}
  </div>
</div>
```

```css
.my-component {
  &__sprite {
    width: 20rem;
  }
}
```

Thus keeping a clear separation of responsibilities between _**my-component**_ and _**sprite**_.

### Review environment Link
https://developers.outreach.psu.edu/footer/?p=viewall-regions-footer

### CODE REVIEW

File: packages/footer/footer.twig [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        </div>
+        {% if colophon %}
+          <div class="footer__colophon">
+            {{ colophon }}
+          </div>
+        {% endif %}
+      {% endset %}
+      {% set footer_bottom_classes = [
+        'footer__bottom-container',
+        bottom_background in ['dark'] ? 'footer__bottom-container--dark',
```
Comment (lal65) [inline packages/footer/footer.twig:95]:
```suggestion
        bottom_background not in ['white'] ? 'footer__bottom-container--dark',
```

File: packages/footer/footer.twig [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      ]|filter(class => class is not empty) %}
+      <div class="{{ footer_bottom_classes|join(' ') }}" data-{{ bottom_background == 'dark' ? 'dark' : 'light' }}>
```
Comment (lal65) [inline packages/footer/footer.twig:97]:
```suggestion
      <div class="{{ footer_bottom_classes|join(' ') }}" data-{{ bottom_background != 'white' ? 'dark' : 'light' }}>
```

File: packages/patternlab/source/patterns/regions/footer/footer~alternate.twig [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{% embed '@psu-ooe/footer/footer.twig' with {
+  bottom_background: 'dark'
```
Comment (lal65) [inline packages/patternlab/source/patterns/regions/footer/footer~alternate.twig:2]: bottom\_background: 'white' -- rename example too.

File: packages/list/list.twig [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{% apply spaceless %}
```
Comment (lal65) [inline packages/list/list.twig:1]: Needs \_list.md docs.

File: packages/physical-address/package.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "@psu-ooe/physical-address",
+  "version": "1.0.0",
+  "description": "Provides an element for physical addresses.",
+  "publishConfig": {
+    "access": "public",
+    "registry": "https://npm.pkg.github.com/"
+  }
```
Comment (lal65) [inline packages/physical-address/package.json:8]: Needs base; oops.

**Review by lal65 — COMMENTED** [review]
I made goof ups :(

File: packages/footer/package.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "@psu-ooe/footer",
+  "version": "1.0.0",
+  "description": "Provides a footer region.",
+  "publishConfig": {
+    "access": "public",
+    "registry": "https://npm.pkg.github.com/"
+  }
+}
```
Comment (kjl16) [inline packages/footer/package.json:9]: Dependency on sprite and background container :) and base?

**Review by kjl16 — CHANGES_REQUESTED** [review]

File: packages/footer/package.json (hunk 6 above)
Comment (lal65) [reply packages/footer/package.json:14]: Tis done.

**Review by kjl16 — APPROVED** [review]

**Pull request merged**